lms/djangoapps/grades/tasks.py

```diff
@@ -186,37 +225,17 @@ def _update_subsection_grades(
                     only_if_higher,
                 )
                 SUBSECTION_SCORE_CHANGED.send(
-                    sender=recalculate_subsection_grade,
+                    sender=None,
                     course=course,
                     course_structure=course_structure,
                     user=student,
                     subsection_grade=subsection_grade,
                 )
 
 
-def _retry_recalculate_subsection_grade(
-        user_id,
-        course_id,
-        usage_id,
-        only_if_higher,
-        expected_modified_time,
-        score_deleted,
-        exc=None,
-):
+def _retry_recalculate_subsection_grade(task_func, exc=None, **kwargs):
     """
     Calls retry for the recalculate_subsection_grade task with the
     given inputs.
     """
-    recalculate_subsection_grade_v2.retry(
-        kwargs=dict(
-            user_id=user_id,
-            course_id=course_id,
-            usage_id=usage_id,
-            only_if_higher=only_if_higher,
-            expected_modified_time=expected_modified_time,
-            score_deleted=score_deleted,
-            event_transaction_id=unicode(get_event_transaction_id()),
-            event_transaction_type=unicode(get_event_transaction_type()),
-        ),
-        exc=exc,
-    )
+    task_func.retry(kwargs=kwargs, exc=exc)
```

Comment:
All this passing around of the calling func makes me think we might be better served by writing our tasks as classes that subclass PersistOnFailureTask instead of using the functional shortcut.